From: Eric Nelson <eric.nelson@boundarydevices.com>
Date: Mon, 12 Mar 2012 15:04:12 +0000 (+0000)
Subject: i.MX6: mx6q_sabrelite: add CONFIG_REVISION_TAG
X-Git-Url: http://git.denx.de/?p=u-boot%2Fu-boot-imx.git;a=commitdiff_plain;h=044cd2a7318a6480346c4e0824956d0dd25e9ec3;hp=b0634306b222cca35ad4754091c718a59f912bc0

i.MX6: mx6q_sabrelite: add CONFIG_REVISION_TAG

This is needed to support Freescale-supplied userspaces.

At the moment, both the IPU and VPU libraries provided by Freescale
in the "imx-lib" package contain routines which scrape the system
revision from /proc/cpuinfo. In the VPU library, this information is
used to load the proper firmware, allowing a single binary to be used
across various i.MX processors.

Signed-off-by: Eric Nelson <eric.nelson@boundarydevices.com>
Acked-by: Marek Vasut <marex@denx.de>
Acked-by: Stefano Babic <sbabic@denx.de>
---

diff --git a/board/freescale/mx6qsabrelite/mx6qsabrelite.c b/board/freescale/mx6qsabrelite/mx6qsabrelite.c
index 7fe2dc9..1d09a72 100644
--- a/board/freescale/mx6qsabrelite/mx6qsabrelite.c
+++ b/board/freescale/mx6qsabrelite/mx6qsabrelite.c
@@ -215,6 +215,11 @@ int board_mmc_init(bd_t *bis)
 }
 #endif
 
+u32 get_board_rev(void)
+{
+	return 0x63000 ;
+}
+
 #ifdef CONFIG_MXC_SPI
 iomux_v3_cfg_t ecspi1_pads[] = {
 	/* SS1 */
diff --git a/include/configs/mx6qsabrelite.h b/include/configs/mx6qsabrelite.h
index bb3a46b..a1a2267 100644
--- a/include/configs/mx6qsabrelite.h
+++ b/include/configs/mx6qsabrelite.h
@@ -33,6 +33,7 @@
 #define CONFIG_CMDLINE_TAG
 #define CONFIG_SETUP_MEMORY_TAGS
 #define CONFIG_INITRD_TAG
+#define CONFIG_REVISION_TAG
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN          (CONFIG_ENV_SIZE + 2 * 1024 * 1024)
